File: string_manipulate/LC344.py
class Solution:
    def reverseString(self, s: list[str]) -> None:
        """
        Do not return anything, modify s in-place instead. O(1) extra memory
        """
        length = len(s)
        for i in range(length // 2):
            s[i], s[length - i - 1] = s[length - i - 1], s[i]


a = Solution()
s = ["h", "e", "l", "l", "o"]
a.reverseString(s)

File: javaalgointerview/LC344.java
public class LC344 {
	class Solution {
		public void reverseString(char[] s) {
			for (int i = 0; i < s.length / 2; i++) {
				swap(s, i, s.length-1-i);
			}
		}

		public void swap(char[] s, int start, int end) {
			char c1 = s[start];
			s[start] = s[end];
			s[end] = c1;
		}
	}
}
